### DISCUSSION TITLE
Stripe Connect Assistance

### DISCUSSION THREAD

**Discussion by wesleypearson** [alfio-event/alf.io#1034] [General]
Could we please get some assistance on getting platform mode set up and enabled?

**Comment by cbellone** [comment]
Hi,

here a step-by-step guide on how to configure it:

1. Read [Stripe Connect's comprehensive docs](https://stripe.com/docs/connect)
2. Head to Stripe Connect settings in your Stripe dashboard
3. when asked to define a Redirect URI enter `https://your-alf.io-instance-url/admin/configuration/payment/stripe/authorize` (replace `your-alf.io-instance-url` with the actual address of your instance)
4. Create a webhook for payment events, as specified here: https://alf.io/docs/configuration/payment/stripe-sca/#configure-stripe-for-strong-customer-authentication-sca 
Remember to select "listen to events on Connected accounts" when creating this webhook
![image](https://user-images.githubusercontent.com/3385346/141470782-3b4250cc-3797-4429-aba8-643b74f5813a.png)
5. Create a webhook for "account.application.deauthorized" with the following URL: `https://your-alf.io-instance-url/api/webhook/stripe/notification`

6. Open Alf.io configuration and head to the General section

![image](https://user-images.githubusercontent.com/3385346/141468874-a4b5d627-68b3-4f84-8777-da93751c8983.png)

Here you can define the fees that you're going to charge to the organizer's stripe account for each transaction, after they connected (see below).
Please note that percentage fee and fixed fee can be combined (e.g. 2% + 0.5 USD **per ticket**)

7. Head to the Stripe Configuration section, **AT SYSTEM LEVEL**

![image](https://user-images.githubusercontent.com/3385346/141470248-9d3a32f0-502b-4a53-bf84-2f6eefc3b947.png)

enter

**Stripe's public key**: the Public API Key of **your account**

**Stripe's secret key**: the Secret Key of **your account**

**Stripe Connect Client ID**: this is the Client ID of your Stripe Connect integration.

![image](https://user-images.githubusercontent.com/3385346/141471700-e73af06c-9da9-4d72-98bb-faa7daa29a3e.png)

**Stripe Connect Callback URL**: overrides the default redirect URI (see above) when connecting a new account. You can leave it blank unless you're planning to have a fleet of alf.io instances under the same Stripe account

**Stripe Signature key for Account-related events**: the webhook signature key from step 5

**Payment Webhook signing secret**: the webhook signature key from step 4

that's basically it. Once you save the configuration, every organization on your alf.io instance will be able to connect their stripe account to yours, and you'll be charging them for each transaction.

![image](https://user-images.githubusercontent.com/3385346/141472412-01f88720-ea47-4777-b9dc-70c58b2f69e4.png)

**Comment by justusweigmann** [comment]
Hey @cbellone - thanks for the detailed Setup Guide. I double-checked if my setup was correct and I don't see any issue in the settings. When I remove the redirect URL from the Stripe checkout that's getting called, it looks like everything works as expected but in the test mode. Could it be that there is a problem updating the right Stripe secret via the settings menu?
{"error":{"message":"Invalid request: Invalid redirect URI 'https://your-alf.io-instance-url/admin/configuration/payment/stripe/authorize'. Ensure this URI exactly matches one of the URIs specified in your application settings"}}
